From 184de8e7227f21de180bb36916abcb316e2986c9 Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Thu, 29 Dec 2022 15:12:03 +0000
Subject: [PATCH 7/9] treble: Securize on-demand

Status is stored in /metadata and controlled by persist prop

Change-Id: I8069b6f471ad87ab34c18b743689ab3584cee35b
---
 phh-prop-handler.sh | 14 ++++++++++++++
 vndk.rc             |  2 ++
 2 files changed, 16 insertions(+)

diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 4371632..a8cea3f 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -210,3 +210,17 @@ if [ "$1" == "persist.sys.phh.disable_soundvolume_effect" ];then
     restartAudio
     exit
 fi
+
+if [ "$1" == "persist.sys.phh.securize" ];then
+    if [[ "$prop_value" != "true" && "$prop_value" != "false" ]]; then
+        exit 1
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        mkdir /metadata/phh
+        touch /metadata/phh/secure
+    else
+        rm /metadata/phh/secure
+    fi
+    exit
+fi
diff --git a/vndk.rc b/vndk.rc
index c150ace..74402a8 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -83,3 +83,5 @@ on property:sys.phh.uninstall-ota=true
 on property:ro.vendor.radio.default_network=*
     setprop ro.telephony.default_network ${ro.vendor.radio.default_network}
 
+on property:persist.sys.phh.securize=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.securize"
-- 
2.34.1

